components/clarinet-format/src/formatter/mod.rs

```diff
@@ -0,0 +1,1232 @@
+                        | DefineFunctions::ReadOnlyFunction
+                        | DefineFunctions::PrivateFunction => format_function(settings, list),
+                        DefineFunctions::Constant | DefineFunctions::PersistedVariable => {
+                            format_constant(settings, list)
+                        }
+                        DefineFunctions::Map => format_map(settings, list, previous_indentation),
+                        DefineFunctions::UseTrait | DefineFunctions::ImplTrait => {
+                            // these are the same as the following but need a trailing newline
+                            format!(
+                                "({})\n",
+                                format_source_exprs(settings, list, previous_indentation)
+                            )
+                        }
+                        // DefineFunctions::Trait => format_trait(settings, list),
+                        // DefineFunctions::PersistedVariable
+                        // DefineFunctions::FungibleToken
+                        // DefineFunctions::NonFungibleToken
+                        _ => format_source_exprs(settings, &[expr.clone()], previous_indentation),
+                    }
+                } else {
+                    to_inner_content(list, settings, previous_indentation)
+                };
+                result.push_str(t(&formatted));
+                continue;
+            }
+        }
+        let current = display_pse(settings, expr, previous_indentation);
+        let mut between = " ";
+        if let Some(next) = iter.peek() {
+            if !is_same_line(expr, next) || is_comment(expr) {
+                between = "\n";
+            }
+        } else {
+            // no next expression to space out
+            between = "";
+        }
+
+        result.push_str(&format!("{current}{between}"));
+    }
+    result
+}
+
+fn format_constant(settings: &Settings, exprs: &[PreSymbolicExpression]) -> String {
+    let func_type = display_pse(settings, exprs.first().unwrap(), "");
+    let indentation = &settings.indentation.to_string();
+    let mut acc = format!("({func_type} ");
+
+    if let Some((name, args)) = name_and_args(exprs) {
+        acc.push_str(&display_pse(settings, name, ""));
+
+        // Access the value from args
+        if let Some(value) = args.first() {
+            if let Some(list) = value.match_list() {
+                acc.push_str(&format!(
+                    "\n{}({})",
+                    indentation,
+                    format_source_exprs(settings, list, "")
+                ));
+                acc.push_str("\n)");
+            } else {
+                // Handle non-list values (e.g., literals or simple expressions)
+                acc.push(' ');
+                acc.push_str(&display_pse(settings, value, ""));
+                acc.push(')');
+            }
+        }
+
+        acc.push('\n');
+        acc.to_owned()
```

Comment:
`acc` is already an owned String. Probably you can simply return `acc` without invoking `.to_owned()`, which avoids extra allocations.